### PR TITLE
#240 Translator should be used inside applications

### PR DESCRIPTION
**Pull request by fkroockmann**
#240

### CODE REVIEW

**Comment by fkroockmann** [comment]
Missing config from rest.

File: specs/tb/component/translator/Translator.spec.js [hunk 1]
```diff
@@ -1,50 +1,63 @@
...
+                expect(Translator.translate('app_only_not_default')).toEqual('app_only_not_default');
+            });
+
+            it("Should log a warning if the translation is not found", function () {
+                spyOn(Logger, "warning").and.returnValue('The key "foo.bar" is malformed.').and.callThrough();
+                Translator.translate('foo.bar');
+                expect(Logger.warning).toHaveBeenCalled();
+            });
+
+            it("Should be able to render key in Renderer scope", function () {
```
Comment (mickaelandrieu) [inline specs/tb/component/translator/Translator.spec.js]: miss a test:

the same with the new ``trans()`` function ;)

File: src/tb/init.js [hunk 2]
```diff
@@ -56,18 +56,26 @@ define(['jquery'], function (jQuery) {
...
 
-                        var initConfig = function () {
+                        var initOnConnect = function () {
                                 DriverHandler.read(self.configUri).done(function (config) {
                                     Core.initConfig(config);
+
+                                    Translator.init({'locale': 'en_US', 'base': 'resources/toolbar/src/tb/i18n/'});
+
+                                    Renderer.addFunction('trans', jQuery.proxy(Translator.translate, Translator));
+
```
Comment (mickaelandrieu) [inline src/tb/init.js]: Also add the filter, please :)

**Comment by ndufreche** [comment]
Ok for me, but really don't like the principle of double assignment for the trans as filter and function.
I think only filter is enough.

**Comment by mickaelandrieu** [comment]
Renderer functions are totaly uncoupled from nunjucks,
Renderer filters are totaly coupled to nunjucks ``but`` as we will probably use the native filters it's consistent to keep them.

Why not let users choose their own ? Functions or Filters implémentation is just few lines of code to maintain.

**Comment by fkroockmann** [comment]
You can merge :)

**Pull request merged**